public/pages/workflow_detail/utils/workflow_to_template_utils.ts

```diff
@@ -62,62 +59,61 @@ function configToProvisionTemplateFlow(config: WorkflowConfig): TemplateFlow {
-function mlProcessorConfigToTemplateNodes(
-  mlProcessorConfig: IProcessorConfig
+// General fn to process all processor configs. Generate a final
+// ingest pipeline containing all of the processors, maintaining order
+function processorConfigsToTemplateNodes(
+  processorConfigs: IProcessorConfig[]
 ): TemplateNode[] {
-  // TODO improvements to make here:
-  // 1. Consideration of multiple ingest processors and how to collect them all, and finally create
-  //    a single ingest pipeline with all of them, in the same order as done on the UI
-  switch (mlProcessorConfig.type) {
-    case PROCESSOR_TYPE.ML:
-    default: {
-      const { model, inputMap, outputMap } = processorConfigToFormik(
-        mlProcessorConfig
-      ) as {
-        model: ModelFormValue;
-        inputMap: MapFormValue;
-        outputMap: MapFormValue;
-      };
-      const ingestPipelineName = generateId('ingest_pipeline');
-
-      let finalProcessor = {
-        ml_inference: {
-          model_id: model.id,
-        },
-      } as MLInferenceProcessor;
-      if (inputMap?.length > 0) {
-        finalProcessor.ml_inference.input_map = inputMap.map((mapEntry) => ({
-          [mapEntry.key]: mapEntry.value,
-        }));
-      }
-      if (outputMap?.length > 0) {
-        finalProcessor.ml_inference.output_map = outputMap.map((mapEntry) => ({
-          [mapEntry.key]: mapEntry.value,
-        }));
-      }
-
-      const finalIngestPipelineDescription =
-        'An ingest pipeline with an ML inference processor.';
-
-      const createIngestPipelineStep = {
-        id: ingestPipelineName,
-        type: WORKFLOW_STEP_TYPE.CREATE_INGEST_PIPELINE_STEP_TYPE,
-        user_inputs: {
-          pipeline_id: ingestPipelineName,
-          model_id: model.id,
-          configurations: {
-            description: finalIngestPipelineDescription,
-            processors: [finalProcessor],
+  const processorsList = [] as IngestProcessor[];
+
+  processorConfigs.forEach((processorConfig) => {
+    // TODO: support more processor types
+    switch (processorConfig.type) {
+      case PROCESSOR_TYPE.ML:
+      default: {
+        const { model, inputMap, outputMap } = processorConfigToFormik(
+          processorConfig
+        ) as {
+          model: ModelFormValue;
+          inputMap: MapFormValue;
+          outputMap: MapFormValue;
+        };
+
+        let processor = {
+          ml_inference: {
+            model_id: model.id,
           },
-        },
-      } as CreateIngestPipelineNode;
-
-      return [createIngestPipelineStep];
+        } as MLInferenceProcessor;
+        if (inputMap?.length > 0) {
+          processor.ml_inference.input_map = inputMap.map((mapEntry) => ({
+            [mapEntry.key]: mapEntry.value,
+          }));
+        }
+        if (outputMap?.length > 0) {
+          processor.ml_inference.output_map = outputMap.map((mapEntry) => ({
+            [mapEntry.key]: mapEntry.value,
+          }));
+        }
```

Comment:
We should keep a separate method for the code under `default` as we might add new processors later.

Reply:
This will change very soon as more processors are added. We will likely not even have a default as an option. Any defaults and what processors we will support is still TBD.

As a reminder this is all in early development still since we have no concrete UX details; just high-level mockups.

Reply:
Note I have a TODO on line 70 already.